### DISCUSSION TITLE
Straightness centrality interpretation - Results Exporting

### DISCUSSION THREAD

**Discussion by mohdalattar** [pysal/momepy#363] [General]
Greetings, 

The main issue of this discussion is that I am skeptical about Straightness centrality results. So I would like you to verify my procedure or if you can explain to me why I am getting these results.

My code is 

```py
import geopandas as gpd
import momepy
import osmnx as ox
import matplotlib.pyplot as plt

G = ox.graph_from_place(“Santa Monica, California, USA”, network_type=“all”)
G_projected = ox.project_graph(G)

edges = ox.graph_to_gdfs(ox.get_undirected(G), nodes=False, edges=True,
                                   node_geometry=False, fill_edge_geometry=True)
primal = momepy.gdf_to_nx(edges, approach=“primal”)
primal = momepy.straightness_centrality(primal)
nodes = momepy.nx_to_gdf(primal, lines=False)
momepy.mean_nodes(primal, “straightness”)
primal_gdf = momepy.nx_to_gdf(primal, points=False)
primal_gdf.to_csv("SM.csv")
ox.save_graph_shapefile(G_projected, filepath=“SantaMonica”) 
```

Then I join the CSV file with the Shapefile from OSMnx.  I am getting the following:

<img width="640" alt="image" src="https://user-images.githubusercontent.com/50058721/173227455-95292cf2-0958-4d80-8f5e-c35145e35516.png">

My questions are:

- I am expecting to have relatively similar straightness centrality score for all grid-like streets. Not sure why this variation in the straigthness centrality. 
- Is there a better way to export the results ? How can I incorporate **ox.set_edge_attributes** function  to export a shapefile with the results of straightness centrality. 

Thanks

**Comment by martinfleis** [comment]
Hi, sorry for a long delay.

A few points.

You have probably noticed this warning.
```
UserWarning: Geometry is in a geographic CRS. Results from 'length' are likely incorrect. Use 'GeoSeries.to_crs()' to re-project geometries to a projected CRS before this operation.
```

Since straightness is based on a distance, you need to make sure you are using the `G_projected` in `ox.graph_to_gdfs`, not `G` as you have in your code.

But even with the code as you have it, the result is this:

![Screenshot 2022-06-22 at 16 57 22](https://user-images.githubusercontent.com/36797143/175062095-e90031c7-323c-413c-a21e-ff31c4f5bdfe.png)

You have used Equal count binning which likely confused you. There's very little variation in fact, apart from that highway in the middle.

> Is there a better way to export the results ?

Yes, you can save it directly to shapefile or any other GIS file format.

```
primal_gdf.to_file("my_file.gpkg")
```
